Purchases/Public/RCPurchasesErrorUtils.m

```diff
@@ -134,6 +142,12 @@ static RCPurchasesErrorCode RCPurchasesErrorCodeFromRCBackendErrorCode(RCBackend
             return RCPurchaseInvalidError;
         case RCBackendEmptyAppUserId:
             return RCInvalidAppUserIdError;
+        case RCBackendInvalidAppleSubscriptionKey:
+            return RCInvalidAppleSubscriptionKeyError;
+        case RCBackendUserIneligibleForPromoOffer:
+            return RCIneligibleError;
```

Comment:
Seems like we can maybe make this more specific?

Reply:
I am down for it. What are you thinking about?